feedly/stream.py

```diff
@@ -148,7 +148,7 @@ def __iter__(self):
 
         while n < self.options._max_count and (self.continuation is not None or self.buffer):
             while self.buffer:
-                i = self.buffer.pop()
+                i = self.buffer.pop(0)
```

Comment:
i think buffer.pop(0) is relatively slow. we should either reverse the buffer on creation or use a data structure that is designed for this kind of operation (from a google search collections.deque would be a good candidate).